stdlib/Option.dora

```diff
@@ -0,0 +1,30 @@
+// FIXME: Identity/Equals/Stringable constraints should all end up on the individual methods that need them, not on the class
+// After nil is gone, this class needs some more intrinsics, so that it can treat <nullptr> as an empty Option (and get rid
+// of the isPresent field). Usual caveats apply: this only works for the first "layer", not for Options inside Options, etc.
+class Option[T : Identity + Equals + Stringable](let value: T, let isPresent: Bool) {
+
+  // self.isPresent.not(): thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0'
+  fun isEmpty() -> Bool = !self.isPresent;
+
+  fun has(rhs: T) -> Bool = self.isPresent && self.value === rhs;
+
+  fun contains(rhs: T) -> Bool =
+    self.isPresent && (self.value === rhs || self.value.equals(rhs));
+
+  fun equals(rhs: Option[T]) -> Bool =
+    self.isPresent == rhs.isPresent && self.value.equals(rhs.value);
+
+  fun toString() -> String {
+    if self.isPresent {
+      return "Some(" + self.value.toString() + ")";
+    } else {
+      return "None";
+    }
+  }
+
+}
+
+// error: no ctor with definition `Option(nil, Bool)`
+// fun none[T : Identity + Equals + Stringable]() -> Option[T] = Option[T](nil, false);
```

Comment:
While this is a bug or at least the error message super misleading, nil isn't compatible with every type. With subclassing we wouldn't have this problem...

Reply:
Dito.

Reply:
Completely forgot about that: for these cases there is `defaultValue()`.

Reply:
Good idea! Done.